src/main/java/Board.java

```diff
@@ -0,0 +1,117 @@
+import java.util.Arrays;
+
+public class Board {
+
+    private Player[][] grid;
+    private int gridSize = 3;
+
+    public Board() {
+        grid = new Player[gridSize][gridSize];
+        for (Player[] row : grid) {
+            Arrays.fill(row, Player.NEITHER);
+        }
+    }
+
+    public Player getSquareValue(int row, int col) {
+        return grid[row][col];
+    }
+
+    public void setSquareValue(int row, int col, Player player) {
+        grid[row][col] = player;
+    }
+
+    public boolean isEmptySquare(int row, int col) {
+        return squareMatches(row, col, Player.NEITHER);
+    }
+
+    public boolean isFull() {
+        for (int row = 0; row < gridSize; row++) {
```

Comment:
My general rule is no more than 2 levels of indentation. The third one here is because of the 2d array :)